assets/js/theme/common/carousel.js

```diff
@@ -2,20 +2,44 @@ import 'slick-carousel';
 
 export default function () {
     const $carousel = $('[data-slick]');
-    if ($carousel.length) {
-        const multipleSlides = $carousel[0].childElementCount > 1;
-        $carousel.slick({ dots: multipleSlides });
-    }
+
+    if ($carousel.length === 0) return;
+
+    const multipleSlides = $carousel[0].childElementCount > 1;
+    $carousel.slick({ dots: multipleSlides });
+
+    const $slidesNodes = $('.heroCarousel-slide');
+
+    $slidesNodes.each((index, element) => {
+        const $element = $(element);
+        const isContentBlock = !!$element.find('.heroCarousel-content').length;
+
+        if (isContentBlock) return true;
+
+        const $image = $element.find('.heroCarousel-image-wrapper img');
+        $('<img/>') // Make in memory copy of image to avoid css issues
+            .attr('src', $($image).attr('src'))
+            .load(function getImageSizes() {
+                const imageRealWidth = this.width; // Note: $(this).width() will not
+                const imageRealHeight = this.height; // work for in memory images.
+
+                const imageAspectRatio = imageRealHeight / imageRealWidth;
+
+                const isImageSquare = imageAspectRatio > 0.8 && imageAspectRatio <= 1.2;
+                const isImageVertical = imageAspectRatio > 1.2;
+
+                const slideClass = isImageSquare && 'is-square-image-type'
+                    || isImageVertical && 'is-vertical-image-type'
+                    || '';
+
+                $element.addClass(slideClass);
+            });
+    });
 
     // Alternative image styling for IE, which doesn't support objectfit
-    if (typeof document.documentElement.style.objectFit === 'undefined') {
-        $('.heroCarousel-slide').each((index, element) => {
-            const $container = $(element);
-            const imgUrl = $container.find('img').data('lazy');
-
-            if (imgUrl) {
-                $container.css('backgroundImage', `url(${imgUrl})`).addClass('compat-object-fit');
-            }
+    if (document.documentElement.style.objectFit === undefined) {
+        $slidesNodes.each((index, element) => {
+            $(element).addClass('compat-object-fit');
```

Comment:
Looks like you could move this code into the above `each` method.

Reply:
I consciously separated this 2 loops. It won't have an influence for performance, but will make code more understandable. You will not have ie11 unnecessary checks in main loop, so it is easier to read. Also the second loop will  work only in ie11 and similar, so form most of browsers in it better even in context of performance. The most expensive operation is finding nodes - that is why I saved it in **$slidesNodes** variable.  But if you insist of course I will combine all logic in one loop